profiler/src/ProfilerEngine/Datadog.Profiler.Native/HeapSnapshotManager.h

```diff
@@ -0,0 +1,183 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+
+#pragma once
+
+#include <thread>
+#include <memory>
+#include <unordered_map>
+#include <chrono>
+
+#include "IHeapSnapshotManager.h"
+#include "IConfiguration.h"
+#include "IFrameStore.h"
+#include "IGarbageCollectionsListener.h"
+#include "IGCDumpListener.h"
+#include "ServiceBase.h"
+#include "MetricsRegistry.h"
+#include "ProxyMetric.h"
+
+#include "corprof.h"
+
+// forward declarations
+class IThreadsCpuManager;
+class INativeThreadList;
+
+using namespace std::chrono_literals;
+
+
+class ClassHistogramEntry
+{
+public:
+    ClassHistogramEntry(std::string& className)
```

Comment:
use `std::string` instead of `std::string&`, you could directly move `className`  into `ClassName`